src/main/scala/io/shiftleft/fuzzyc2cpg/Fuzzyc2Cpg.scala

```diff
@@ -0,0 +1,139 @@
+package io.shiftleft.fuzzyc2cpg
+
+import java.nio.file.Path
+
+import io.shiftleft.codepropertygraph.generated.Languages
+import io.shiftleft.fuzzyc2cpg.Utils.{getGlobalNamespaceBlockFullName, newEdge, newNode}
+import io.shiftleft.fuzzyc2cpg.output.protobuf.OutputModuleFactory
+import io.shiftleft.proto.cpg.Cpg.{CpgStruct, NodePropertyName}
+import io.shiftleft.proto.cpg.Cpg.CpgStruct.Edge.EdgeType
+import io.shiftleft.proto.cpg.Cpg.CpgStruct.Node
+import io.shiftleft.proto.cpg.Cpg.CpgStruct.Node.NodeType
+import org.slf4j.LoggerFactory
+import io.shiftleft.fuzzyc2cpg.Utils._
+import io.shiftleft.fuzzyc2cpg.output.CpgOutputModuleFactory
+import io.shiftleft.fuzzyc2cpg.parser.modules.AntlrCModuleParserDriver
+
+class Fuzzyc2Cpg[T](outputModuleFactory : CpgOutputModuleFactory[T]) {
+
+  def this(outputPath : String) = {
+    this(new OutputModuleFactory(outputPath, true, false)
+      .asInstanceOf[CpgOutputModuleFactory[T]])
+  }
+
+  def runAndOutput(fileAndDirNames: Array[String]) = {
+    val inputPaths = fileAndDirNames
+    val sourceFileNames = SourceFiles.determine(inputPaths.toList).sorted
+
+    val filenameToNamespaceBlock = createStructuralCpg(sourceFileNames, outputModuleFactory)
+
+    // TODO improve fuzzyc2cpg namespace support. Currently, everything
+    // is in the same global namespace so the code below is correctly.
+    filenameToNamespaceBlock.par.foreach(createCpgForCompilationUnit)
```

Comment:
Nice 👏 General question: does `par` create separate JVM threads for each namespace (currently `<global>` only). I am trying to understand, is it one thread per file for now? In that case, will this lead to contention because of high scheduling load (hundreds of threads trying to be scheduled per core)?

Reply:
no, not one thread per file. There will be worker threads and the load is distributed over the worker threads.